k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -2020,10 +2026,11 @@ void processPendingExpunge(PendingExpunge command, Account account) throws Messa
             if (remoteFolder.getMode() != Folder.OPEN_MODE_RW) {
                 return;
             }
-            remoteFolder.expunge();
+            remoteFolder.expunge(localFolder.getAllMessageUids(true));
```

Comment:
This is a manually triggered expunge and should therefore unconditionally expunge messages, i.e. call `expunge()` without arguments.